web/src/main/java/org/fao/geonet/services/metadata/MetadataCacheSeeder.java

```diff
@@ -0,0 +1,397 @@
+ *   stored in the catalog,
+ * 
+ * - it maintains a sitemap.xml file for better indexation against search engines ;
+ * 
+ * To function properly, it has to be configured (defined as bean) into config-spring-geonetwork.xml, using the following
+ * snippet:
+ * 
+ *     <bean name="metadataCacheSeeder" class="org.springframework.scheduling.quartz.JobDetailFactoryBean">
+ *       <property name="jobClass" value="org.fao.geonet.services.metadata.MetadataCacheSeeder" />
+ *       <property name="jobDataAsMap">
+ *         <map>
+ *           <entry key="xslStylesheet" value="pigma-static-html" />
+ *           <entry key="cachePath" value="/tmp/pigma-md-cache" />
+ *         </map>
+ *       </property>
+ *   </bean>
+ *
+ *   <bean id="metadataCacheSeederCronTrigger" class="org.springframework.scheduling.quartz.CronTriggerFactoryBean">
+ *       <property name="jobDetail" ref="metadataCacheSeeder" />
+ *       <!-- runs every days at 4:00 AM -->
+ *       <property name="cronExpression" value="0 0 4 * * ?" />
+ *       <property name="startDelay" value="30000"/>
+ *   </bean>
+ *   
+ *   <bean class="org.springframework.scheduling.quartz.SchedulerFactoryBean">
+ *       <property name="jobFactory">
+ *           <bean class="org.fao.geonet.util.spring.AutowiringSpringBeanJobFactory"/>
+ *       </property>
+ *       <property name="triggers">
+ *           <list>
+ *               <ref bean="metadataCacheSeederCronTrigger" />
+ *           </list>
+ *       </property>
+ *   </bean>
+ *
+ * @author pmauduit
+ *
+ */
+public class MetadataCacheSeeder extends QuartzJobBean implements ApplicationContextAware {
+
+    /**
+     * The path on the file system where the files can be copied.
+     */
+    private String cachePath;
+    
+    /**
+     * The XSL stylesheet to be used, in the formatter context,
+     * basically the name of the directory containing the view.xsl stylesheet.
+     */
+    private String xslStylesheet;
+
+    /**
+     * The full path to the XSL stylsheet.
+     */
+    private String xslPath;
+    
+    /**
+     * The prefix where the files would be publicly available.
+     * (e.g. http://ids.pigma.org/mds for http://ids.pigma.org/mds/[uuid].html)
+     */
+    private String urlPath = "http://ids.pigma.org/mds";
+    
+    /**
+     * Default change frequency for the sitemap items.
+     */
+    private String changeFreq = "daily";
+    /**
+     * Flag to control if the process is currently running.
+     */
+    private static AtomicBoolean isRunning = new AtomicBoolean(false);
+    
+    /**
+     * The namespace to be used for generating the sitemap.
+     */
+    private final String SITEMAP_NS = "http://www.google.com/schemas/sitemap/0.84";
+    
+    
+    /**
+     * The spring application context, used to get a hook to necessary beans of the running catalogue.
+     */
+    private ApplicationContext applicationContext;
```

Comment:
This class is a bean so you should be able to add `@Autowired` to this field and have spring set it for you